types/model.go

```diff
@@ -9,91 +9,101 @@ type FunctionDeployment struct {
 
 	// EnvVars provides overrides for functions.
-	EnvVars map[string]string `json:"envVars"`
-
-	// RegistryAuth is the registry authentication (optional)
-	// in the same encoded format as Docker native credentials
-	// (see ~/.docker/config.json)
-	RegistryAuth string `json:"registryAuth,omitempty"`
+	EnvVars map[string]string `json:"envVars,omitempty"`
 
 	// Constraints are specific to back-end orchestration platform
-	Constraints []string `json:"constraints"`
+	Constraints []string `json:"constraints,omitempty"`
 
 	// Secrets list of secrets to be made available to function
-	Secrets []string `json:"secrets"`
+	Secrets []string `json:"secrets,omitempty"`
 
 	// Labels are metadata for functions which may be used by the
 	// back-end for making scheduling or routing decisions
-	Labels *map[string]string `json:"labels"`
+	Labels *map[string]string `json:"labels,omitempty"`
 
 	// Annotations are metadata for functions which may be used by the
 	// back-end for management, orchestration, events and build tasks
-	Annotations *map[string]string `json:"annotations"`
+	Annotations *map[string]string `json:"annotations,omitempty"`
 
 	// Limits for function
-	Limits *FunctionResources `json:"limits"`
+	Limits *FunctionResources `json:"limits,omitempty"`
 
 	// Requests of resources requested by function
-	Requests *FunctionResources `json:"requests"`
+	Requests *FunctionResources `json:"requests,omitempty"`
 
 	// ReadOnlyRootFilesystem removes write-access from the root filesystem
 	// mount-point.
-	ReadOnlyRootFilesystem bool `json:"readOnlyRootFilesystem"`
+	ReadOnlyRootFilesystem bool `json:"readOnlyRootFilesystem,omitempty"`
+}
 
-	// Namespace for the function to be deployed into
+// Secret for underlying orchestrator
+type Secret struct {
+	Name      string `json:"name"`
 	Namespace string `json:"namespace,omitempty"`
+	Value     string `json:"value,omitempty"`
 }
 
 // FunctionResources Memory and CPU
 type FunctionResources struct {
-	Memory string `json:"memory"`
-	CPU    string `json:"cpu"`
+	Memory string `json:"memory,omitempty"`
+	CPU    string `json:"cpu,omitempty"`
 }
 
 // FunctionStatus exported for system/functions endpoint
 type FunctionStatus struct {
-
-	// Name corresponds to a Service
-	Name string `json:"name"`
+	// Service corresponds to a Service
+	Service string `json:"service"`
 
 	// Image corresponds to a Docker image
 	Image string `json:"image"`
 
-	// InvocationCount count of invocations
-	InvocationCount float64 `json:"invocationCount"`
+	// Namespace for the function to be deployed into
+	Namespace string `json:"namespace,omitempty"`
 
-	// Replicas desired within the cluster
-	Replicas uint64 `json:"replicas"`
+	// EnvProcess corresponds to the fprocess variable for your container watchdog.
+	EnvProcess string `json:"envProcess,omitempty"`
 
-	// EnvProcess is the process to pass to the watchdog, if in use
-	EnvProcess string `json:"envProcess"`
+	// EnvVars provides overrides for functions.
```

Comment:
this command doesn't seem right, or at least isn't what i expected for "EnvVars"

Reply:
I don't follow. Can you explain what you mean another way?

Reply:
`EnvProcess` was there already, `EnvVars` is new